code/game/objects/machinery/computer/nuke_disk_generator.dm

```diff
@@ -0,0 +1,169 @@
+// -- Print disk computer
+/obj/item/circuitboard/computer/nuke_disk_generator
+	name = "circuit board (nuke disk generator)"
+	build_path = /obj/machinery/computer/nuke_disk_generator
+
+
+/obj/machinery/computer/nuke_disk_generator
+	name = "nuke disk generator"
+	desc = "Used to generate the correct auth discs for the nuke."
+	icon_state = "request" // TODO: Better icon state
+	circuit = "/obj/item/circuitboard/computer/nuke_disk_generator"
+
+	resistance_flags = INDESTRUCTIBLE|UNACIDABLE
+
+	var/generate_time = 1.5 MINUTES // time for the machine to generate the disc
+	var/segment_time = 15 SECONDS // time to start the hack 
+	var/printing_time = 15 SECONDS // time to print a disk 
+
+	var/total_segments = 5 // total number of times the hack is required
+	var/completed_segments = 0 // what segment we are on, (once this hits total, disk is printed)
+	var/current_timer
+
+	var/reprintable = FALSE // once the disk is printed, reprinting is enabled
+	var/printing = FALSE // check if someone is printing already
+
+	var/disk_type
+	var/obj/item/disk/nuclear/disk
+
+	var/list/technobabble = list(
+		"Booting up terminal-  -Terminal running",
+		"Establishing link to offsite mainframe- Link established",
+		"WARNING, DIRECTORY CORRUPTED, running search algorithms- nuke_fission_timing.exe found",
+		"Invalid credentials, upgrading permissions through TGMC military override- Permissions upgraded, nuke_fission_timing.exe available",
+		"Downloading nuke_fission_timing.exe to removable storage- nuke_fission_timing.exe downloaded to floppy disk, have a nice day"
+	)
+
+/obj/machinery/computer/nuke_disk_generator/Initialize()
+	. = ..()
+
+	if(!disk_type)
+		WARNING("disk_type is required to be set before init")
+		return INITIALIZE_HINT_QDEL
+
+	GLOB.nuke_disk_generators += src
+
+
+/obj/machinery/computer/nuke_disk_generator/Destroy()
+	GLOB.nuke_disk_generators -= src
+	return ..()
+
+
+/obj/machinery/computer/nuke_disk_generator/process()
+	. = ..()
+	if(. || !current_timer)
+		return
+
+	deltimer(current_timer)
+	current_timer = null

```

Comment:
could return PROCESS_KILL so it doesn't waste another tick

Reply:
Done.